osutil/user.go

```diff
@@ -126,13 +126,20 @@ func AddUser(name string, opts *AddUserOptions) error {
 	return nil
 }
 
+var userCurrent = user.Current
+
 // RealUser finds the user behind a sudo invocation, if applicable and possible.
 func RealUser() (*user.User, error) {
-	cur, err := user.Current()
+	cur, err := userCurrent()
 	if err != nil {
 		return nil, err
 	}
 
+	// not root, so no sudo invocation we care about
+	if cur.Uid != "0" {
+		return cur, nil
+	}
+
```

Comment:
I found this comment confusing. Can you change the comment for the function to be:

```
// RealUser finds the user behind a sudo invocation when root, if applicable and possible.
```

and then change this comment to be:
```
// Don't check SUDO_USER when not root and simply return the current uid
// to properly support sudo'ing from root to a non-root user
```

Note that this change makes sense but I wonder if RealUser now is the right name for the function. I don't have a good alternative otoh though...